homeassistant/data_entry_flow.py

```diff
@@ -100,20 +100,31 @@ class FlowResult(TypedDict, total=False):
     reason: str
     required: bool
     result: Any
-    step_id: str
     title: str
     type: FlowResultType
     url: str
     version: int
 
 
+class FlowResult(_BaseFlowResult, total=False):
+    """Typed result dict."""
+
+    step_id: str
```

Comment:
Since `total=False`, `step_id` can still be `None`.

Whishing these where dataclasses 😞

Reply:
`total=False` means the key can be missing from the dictionary, not that its type can be another type

Reply:
Yeah, sorry. Good morning. I meant that.

So, indeed, `step_id` cannot be `None`, but its existence is not guaranteed. Therefore, the checks that a user of this `FlowResult` needs to do, is basically the same for both cases.

`if step_id := result.get("step_id"):`

The only thing this one adds is the possibility to check existence first and next access it directly.

```
if "step_id" in result:
    result["step_id"]
```

It doesn't provide much "win".

Essentially, a dictionary is not the right datatype to use here.